src/CFBG.cpp

```diff
@@ -126,75 +126,238 @@ uint32 CFBG::GetAllPlayersCountInBG(Battleground* bg)
         }
-        else if (player->getGender() == GENDER_MALE) // HORDE PLAYER, ONLY HAVE MALE Night Elf ID
-        {
-            FakeMorph = FAKE_M_NELF;
-            FakeRace = RACE_NIGHTELF;
+    } else {
+        // FEMALE races DWARF, NIGHT ELF, TROLL are missing
+        switch(race) {
+            case RACE_ORC:
+                return FAKE_F_ORC;
+            case RACE_DRAENEI:
+                return FAKE_F_DRAENEI;
+            case RACE_HUMAN:
+                return FAKE_F_HUMAN;
+            case RACE_BLOODELF:
+                return FAKE_F_BLOOD_ELF;
+            case RACE_GNOME:
+                return FAKE_F_GNOME;
+            case RACE_TAUREN:
+                return FAKE_F_TAUREN;
+            default:
+                return FAKE_F_BLOOD_ELF; // this should never happen, it to fix a warning about return value
         }
-        else
-            FakeRace = player->GetTeamId(true) == TEAM_ALLIANCE ? RACE_BLOODELF : RACE_HUMAN;
+    }
+}
 
-        if (player->GetTeamId(true) == TEAM_HORDE)
-        {
-            if (player->getGender() == GENDER_MALE)
-                FakeMorph = 19723;
-            else
-                FakeMorph = 19724;
+
+void CFBG::randomRanceMorph(uint8* race, uint32* morph, TeamId team, uint8 _class, uint8 gender) {
+
+    // if alliance find a horde race
+    if (team == TEAM_ALLIANCE) {
+
+        // default race because UNDEAD morph is missing
+        *race = RACE_BLOODELF;
+
+        if (gender == GENDER_MALE) {
+            *morph = FAKE_M_BLOOD_ELF;
+
+            switch (_class) {
+                case CLASS_DRUID:
+                    *race = RACE_TAUREN;
+                    *morph = FAKE_M_TAUREN;
+                    break;
+                case CLASS_SHAMAN:
+                case CLASS_WARRIOR:
+                    *race = getRandomRace({ RACE_ORC, RACE_TAUREN, RACE_TROLL });
+                    *morph = getMorphFromRace(*race, gender);
+                    break;
+                case CLASS_PALADIN:
+                    // BLOOD ELF, so default race
+                    break;
+                case CLASS_HUNTER:
+                case CLASS_DEATH_KNIGHT:
+                    *race = getRandomRace({ RACE_ORC, RACE_TAUREN, RACE_TROLL, RACE_BLOODELF });
+                    *morph = getMorphFromRace(*race, gender);
+                    break;
+                case CLASS_ROGUE:
+                    *race = getRandomRace({ RACE_ORC, RACE_TROLL, RACE_BLOODELF });
+                    *morph = getMorphFromRace(*race, gender);
+                    break;
+                case CLASS_MAGE:
+                case CLASS_PRIEST:
+                    *race = getRandomRace({ RACE_TROLL, RACE_BLOODELF });
+                    *morph = getMorphFromRace(*race, gender);
+                    break;
+                case CLASS_WARLOCK:
+                    *race = getRandomRace({ RACE_ORC, RACE_BLOODELF });
+                    *morph = getMorphFromRace(*race, gender);
+                    break;
```

Comment:
add a TODO with the missing ones

Reply:
I do not add TODO, because we can't do this for what we have in the client-side
Anyway I added for all classes what race is missing, so in future, if we solve this in some way, we can just read and replace the missing races with some new morphs/models.